bolt-cli/src/commands/delegate/web3signer.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{
+    cli::{Action, Web3SignerOpts},
+    commands::delegate::types::{
+        DelegationMessage, RevocationMessage, SignedDelegation, SignedRevocation,
+    },
+    common::web3signer::Web3Signer,
+};
+use ethereum_consensus::crypto::{PublicKey as BlsPublicKey, Signature as BlsSignature};
+use eyre::Result;
+use tracing::debug;
+
+use super::types::SignedMessage;
+
+/// Generate signed delegations/recovations using a remote Web3Signer.
+pub async fn generate_from_web3signer(
+    opts: Web3SignerOpts,
+    delegatee_pubkey: BlsPublicKey,
+    action: Action,
+) -> Result<Vec<SignedMessage>> {
+    // Connect to web3signer.
+    let mut web3signer = Web3Signer::connect(opts.url, opts.tls_credentials).await?;
+
+    // Read in the accounts from the remote keystore.
+    let accounts = web3signer.list_accounts().await?;
+    debug!("Found {} remote accounts to sign with", accounts.len());
+
+    let mut signed_messages = Vec::with_capacity(accounts.len());
+
+    for account in accounts {
+        // Parse the BLS key of the account.
+        // Trim the pre-pended 0x.
+        let trimmed_account = &account.clone()[2..];
```

Comment:
This is a bit more graceful compared to panicking upon out of bounds unexpectedly.
```suggestion
        let trimmed_account =
            account.get(2..).ok_or_else(|| eyre::eyre!("Invalid account: {account}"))?;
```
The same applies below with signatures, if you want to can add a little util for that!

Reply:
Will add - good catch.

Reply:
Updated 🫡  https://github.com/chainbound/bolt/pull/513/commits/690f469ed16cdee8794263694738fb15790cab4b